apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/DraggableDrawerButton.tsx

```diff
@@ -6,81 +6,117 @@ import type {
   draggableProps: DraggableProvidedDraggableProps
   dragHandleProps: DraggableProvidedDragHandleProps | null
   setSelectedIndex: (selectedIndex?: number) => void
+  isError: boolean
 }
 
 const DraggableDrawerButtonText = withJsonFormsMasterListItemProps(
   ({ childLabel, index }: StatePropsOfMasterItem) => (
-    <Text
-      textStyle="subhead-2"
-      textColor="base.content.strong"
-      textAlign="start"
-    >
+    <Text textStyle="subhead-2" textAlign="start">
       {childLabel || `Item ${index + 1}`}
     </Text>
   ),
 )
 
-const DraggableDrawerButton = forwardRef(
+const DraggableDrawerButton = forwardRef<DraggableDrawerButtonProps, "div">(
   (
     {
       draggableProps,
       dragHandleProps,
       setSelectedIndex,
       index,
+      isError,
       ...rest
-    }: DraggableDrawerButtonProps,
+    },
     ref,
   ) => {
     return (
-      <Box
-        my="0.375rem"
-        borderWidth="1px"
-        borderColor="base.divider.medium"
-        borderRadius="lg"
-        w="100%"
-        bgColor="utility.ui"
-        transitionProperty="background-color"
-        transitionDuration="0.2s"
-        _hover={{
-          bgColor: "interaction.main-subtle.default",
-        }}
-        _active={{
-          bgColor: "interaction.main-subtle.default",
-          borderColor: "base.content.brand",
-          outline: "1px solid",
-          outlineColor: "base.content.brand",
-        }}
-        ref={ref}
-        {...draggableProps}
-      >
-        <HStack spacing={0}>
+      <Box my="0.375rem" ref={ref} {...draggableProps} w="full">
+        <HStack
+          spacing={0}
+          border="1px solid"
+          borderColor="base.divider.medium"
+          borderRadius="6px"
+          bg="white"
+          transitionProperty="common"
+          transitionDuration="normal"
+          aria-invalid={isError}
+          _hover={{
+            bg: "interaction.muted.main.hover",
+            borderColor: "interaction.main-subtle.hover",
+            _invalid: {
+              bg: "interaction.muted.critical.hover",
+              borderColor: "utility.feedback.critical",
+            },
+          }}
+          _active={{
+            bg: "interaction.main-subtle.default",
+            borderColor: "interaction.main-subtle.hover",
+            shadow: "0px 1px 6px 0px #1361F026",
+            _invalid: {
+              bg: "interaction.muted.critical.hover",
+              borderColor: "utility.feedback.critical",
+              shadow: "0px 1px 6px 0px #C0343426",
+            },
+          }}
+          align="stretch"
+          overflow="hidden"
+        >
+          {isError && (
+            <Box
+              aria-hidden
+              bg="utility.feedback.critical"
+              width="6px"
+              mr="-6px"
+            />
+          )}
           <Box
             cursor="grab"
             pl="0.5rem"
             pr="0.25rem"
-            py="0.75rem"
+            py="1.25rem"
```

Comment:
issue: the drag handle doesn't seem to be middle aligned

<img width="288" alt="image" src="https://github.com/user-attachments/assets/0a23e956-4645-4376-8f84-70e1163a925b">